### DISCUSSION TITLE
关于industrialcraft-2-2.9.36+ex119-1.19.2 在服务器上运行的错误问题

### DISCUSSION THREAD

**Discussion by Lu-Yanzhi** [IndustrialCraftMC/IndustrialCraft2#144] [【漏洞反馈】Bugs Report]
Hello,

I have the trouble when I enter the server as a player. I also can not enter the server.

Server: Minecraft 1.19.2 with Fabric.
Mod: Fabric 0.15.4 , industrialcraft-2-2.9.36+ex119-1.19.2-fabric.

Client:Minecraft 1.19.2 with Fabric.
Mod: Fabric 0.15.4 , industrialcraft-2-2.9.36+ex119-1.19.2-fabric.

Client-Side Error Message:
Internal Exception: io.netty.handler.codec DecoderException: io.netty.handler
codec.EncoderException:java.io.IOException: Root tag must be a named compound tag.

Server-Side Logs:
[13:42:50] [Server thread/INFO]: Lu_Yanzhi joined the game
[13:42:50] [Server thread/INFO]: Lu_Yanzhi lost connection: Disconnected
[13:42:50] [Server thread/INFO]: Lu_Yanzhi left the game

Thank you for your time and assistance.

你好，
       我这里目前有一个bug是关于工业2 mod在服务器端运行的问题，目前我在我的服务器端和客户端上仅仅安装了工业2 和Fabric ，但我在客户端无法登陆服务器，产生错误：

客户端与服务端：Minecraft 1.19.2 with Fabric.
Mod：Fabric 0.15.4 , industrialcraft-2-2.9.36+ex119-1.19.2-fabric.

客户端错误信息：
Internal Exception: io.netty.handler.codec DecoderException: io.netty.handler
codec.EncoderException:java.io.IOException: Root tag must be a named compound tag.
服务器端信息：
[13:42:50] [Server thread/INFO]: Lu_Yanzhi joined the game
[13:42:50] [Server thread/INFO]: Lu_Yanzhi lost connection: Disconnected
[13:42:50] [Server thread/INFO]: Lu_Yanzhi left the game

感谢您的阅读，期待您的回复！

**Comment by nkdns** [comment]
https://github.com/IndustrialCraftMC/IndustrialCraft2/issues/129
issues中提出了同样的问题，我们基于最新的Fabric和Forge都进行了尝试，均报告配方错误，导致玩家无法加入服务器，在mod文件夹中删除报错配方之后会换一个配方报错，疑似是配方挂载出现了错误，希望作者可以快点修复

**Comment by RainbowXXX** [comment]
这个问题是由于配方序列化器导致的，在作者修复bug之前可以先使用我的[mod](https://github.com/HM-GDTR/IC2-Recipe-FIX)来临时修复这个bug.

**Comment by qyl27** [comment]
这是一个配方反序列化问题，似乎已经存在很久了？但模组并未开源，没法PR修复。
See also #145 .
